### PR TITLE
docs: add try job labels to CONTRIBUTING

### PR DESCRIPTION
**Pull request by delan**
This patch documents the new try job labels added in #30383 and #30712.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] ~~These changes fix #___ (GitHub issue number if applicable)~~

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they are in the documentation

### CODE REVIEW

File: CONTRIBUTING.md
Comment (mrobinson) [inline CONTRIBUTING.md:28]: I think this is a difference between North America and Europe title-casing: https://www.uoc.edu/portal/en/servei-linguistic/convencions/british-american-english/capitalization/index.html

I guess we should standardize on one of these styles.

File: CONTRIBUTING.md
Comment (mrobinson) [inline CONTRIBUTING.md:69]: Would it be okay to hold off on this a bit? The hope is that the labels (which we're still testing) are good enough to remove the comment style entirely, but only if the GitHub Actions works better. The motivation is that there are quite a few issues with the GitHub Action that is run for every PR comment (lots of spurious failures).

Comment (delan) [reply CONTRIBUTING.md:69]: sure, no worries! closing for now, we can revisit this later :)

File: CONTRIBUTING.md
Comment (delan) [reply CONTRIBUTING.md:28]: haha, i think we’ve run into this before. regional conventions aside, i think title case should be avoided because it destroys the details and nuance conveyed by capitalisation.

File: CONTRIBUTING.md
Comment (delan) [reply CONTRIBUTING.md]: now that #30712 is merged, how does this look with 9b28c527e6534afdfb40691f7b72d7135d34bb17?

File: CONTRIBUTING.md
Comment (mrobinson) [inline CONTRIBUTING.md:54]:
```suggestion
When you push to a pull request, GitHub automatically checks that your changes have no compilation, lint, or tidy errors.
```

File: CONTRIBUTING.md
Comment (mrobinson) [inline CONTRIBUTING.md:56]: Only those with write access to the repository can add labels, so it might be worth adding sentence here explaining what to do if you cannot add a label.

```suggestion
To run unit tests or Web Platform Tests against a pull request, add one or more of the labels below to your pull request. If you do not have permission to add labels to your pull request, add a comment on your bug requesting that they be added.
```

**Review by mrobinson — APPROVED** [review]
Thanks for the wait. This look good with a couple minor suggestions.

File: CONTRIBUTING.md
Comment (delan) [reply CONTRIBUTING.md]: done in 6065670293c4992ccc0661c1a245ea84e5b3daa8

File: CONTRIBUTING.md
Comment (delan) [reply CONTRIBUTING.md]: done in fddc8b2b8e30daa09ca92f435bedece7770fe8bb

**Pull request merged**
